### PR TITLE
fix update check

### PR DESCRIPTION
**Pull request by hlarsen**
fixed version check on update, also abort if passed a nonexistent version and added usage docs.

### CODE REVIEW

File: lib/boxcar/command/update.rb [hunk 1]
```diff
@@ -61,9 +69,13 @@ def index
 
 private
 
-  def latest_stable
```
Comment (nicinabox) [inline lib/boxcar/command/update.rb:64]: Let's keep `latest_stable` as an alias to `all_versions.first`. It's a bit clearer that way.

File: lib/boxcar/command/update.rb [hunk 2]
```diff
@@ -61,9 +69,13 @@ def index
...
+  def all_versions
     response = self.class.get('https://api.github.com/repos/nicinabox/boxcar/tags')
     tags = JSON.parse(response.body)
-    tags.first[0].name
```
Comment (nicinabox) [inline lib/boxcar/command/update.rb:67]: Be sure to merge from upstream. This line was changed in a later commit.

File: lib/boxcar/command/update.rb [hunk 3]
```diff
@@ -6,9 +6,17 @@
...
+  # Update Boxcar
+  # update <optional-version>
+  #
+  # If version is not specified, latest stable will be used
   def index
-    version = args.first || latest_stable
-    abort "You're on the latest stable" unless version.present?
+    version = args.first || all_versions[0]
+    puts "Latest stable version: #{all_versions[0]}"
+    abort "Requested version (#{version}) not found" unless all_versions.include? version
```
Comment (nicinabox) [inline lib/boxcar/command/update.rb:18]: This is great :+1:

**Comment by nicinabox** [comment]
@hlarsen I merged and refactored a bit. [Have a look.](https://github.com/nicinabox/boxcar/blob/hlarsen-master/lib/boxcar/command/update.rb). I need to test when I get home (or you can), but I think this should work.

**Comment by hlarsen** [comment]
the unless on line23 isn't working as written; works if it's two separate includes for each array, though.

also looks like there is an issue while fetching:  mv: cannot stat `build/usr/apps/boxcar/boot': No such file or directory

closing this pull request; need to do some more reading on branches/forking/etc to be sure i'm doing it all "the right way"

**Comment by nicinabox** [comment]
Oh yeah, I've moved that boot script to event. I need to setup the build mapping for that.

**Pull request closed without merging**